src/sdk/models/platformCompare/rules.ts

```diff
@@ -8,50 +8,118 @@ import {
 	ifSizeGreaterThan,
 	checkFeatures,
 	checkPath,
-	availableIfSupported
+	availableIfSupported,
+	ifFeaturePresentNotCompatible
 } from './util';
 
 const comparisonRules: PlatformComparisonRuleGroup[] = [
+	{
+		id: 'versions',
+		name: 'File versions',
+		available: (left, right) => availableIfSupported(left, right, 'features.versioning'),
+		rules: [
+			{
+				id: 'version_preservation',
+				name: 'Version preservation',
+				compare: checkFeatures((left, right) => ifFeaturePresent(left, right, 'versioning'))
+			}
+		]
+	},
+	{
+		id: 'lock_propagation',
+		name: 'Lock propagation',
+		available: (left, right) => availableIfSupported(left, right, 'features.checkout'),
+		rules: [
+			{
+				id: 'file_lock_propagation',
+				name: 'File lock propagation',
+				compare: checkFeatures((left, right) => ifFeaturePresent(left, right, 'checkout'))
+			},
+			{
+				id: 'mirror_lock_owner',
+				name: 'Mirror lock ownership',
+				compare: checkFeatures((left, right) => ifFeaturePresent(left, right, 'checkout.set_owner'))
+			}
+		]
+	},
 	{
 		id: 'validation',
 		name: 'Path validation',
+		available: (left, right) => availableIfSupported(left, right, 'path.validation'),
 		rules: [
+			{
+				id: 'invalid_characters',
+				name: 'Invalid characters',
+				compare: checkPath((left, right) => ifFeaturePresentNotCompatible(left, right, 'invalid_characters'))
```

Comment:
Shouldn't this use `ifStringArrayExists`? I assume that is what the function is for and if the two are present but compatible (i.e. invalid characters from source are all considered invalid on destination) then we should show it as such.

Reply:
Yes. The intention is to use isStringArrayExists; however, I was running into an issue loading/implementing the `lodash.difference` function. Thought was to split out that functionality, and implement basic functionality as a stop gap for the code freeze.